applications/tari_base_node/src/builder.rs

```diff
@@ -230,16 +230,33 @@ pub async fn configure_and_initialize_node(
     node_identity: Arc<NodeIdentity>,
     wallet_node_identity: Arc<NodeIdentity>,
     interrupt_signal: ShutdownSignal,
+    cleanup_orphans_at_startup: bool,
 ) -> Result<BaseNodeContext, anyhow::Error>
 {
     let result = match &config.db_type {
         DatabaseType::Memory => {
             let backend = MemoryDatabase::<HashDigest>::default();
-            build_node_context(backend, node_identity, wallet_node_identity, config, interrupt_signal).await?
+            build_node_context(
+                backend,
+                node_identity,
+                wallet_node_identity,
+                config,
+                interrupt_signal,
+                cleanup_orphans_at_startup,
```

Comment:
Its already in the config? Rather than passing in a bool, you can just call `config. orphan_db_clean_out_threshold>0` where you need to pass it into the db.

Reply:
No, `GlobalConfig` has an `orphan_db_clean_out_threshold` member that is fed in form the config file, which is different than `cleanup_orphans_at_startup` that is fed in from the command line.